orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/pipeline/BakeStage.groovy

```diff
@@ -129,7 +157,31 @@ class BakeStage implements StageDefinitionBuilder {
           return deploymentDetails
         }
       ]
+
+      if (failOnImageNameMismatchEnabled() && !stage.context.dontFailOnImageNameMismatch) {
+         List<String> distinctImageNames = globalContext.deploymentDetails.stream()
+          .map({details -> details['imageName']})
+          .distinct()
+          .collect(Collectors.toList())
+
+        Joiner joiner = Joiner.on(", ")
+
+        if (distinctImageNames.size() > 1) {
+          throw new ConstraintViolationException(
+            "Image names found in different regions do not match: ${joiner.join(distinctImageNames)}. "
+            + "Re-run the bake to protect against deployment failures.")
+        }
+      }
+
       TaskResult.builder(ExecutionStatus.SUCCEEDED).outputs(globalContext).build()
     }
+
+    private boolean failOnImageNameMismatchEnabled() {
+      try {
+        return dynamicConfigService.isEnabled("stages.bake.failOnImageNameMismatch", false)
+      } catch (Exception e) {
```

Comment:
Are you sure swallowing any exception without logging is a good idea?

Reply:
I would generally ask the same question, but the code @marchello2000 and I looked at as an example of using `DynamicConfigService` looked like this:
https://github.com/spinnaker/orca/blob/fba04ec8cfc5e9c8d5d0cb6219b2507822ae2358/orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/StageDefinitionBuilder.java#L106-L120

...which is obviously not an excuse, but the method doesn't declare any exceptions, and I was told a common practice is to throw descendants of `RuntimeException`, so I made the incorrect assumption that returning the default when anything went wrong was OK.

I see now that most calls are actually not within a `try/catch` block. Should I just remove them?

Reply:
Definitely not the correct approach: Just because a bad pattern is done somewhere else doesn't mean we should spread it elsewhere! We don't do checked exceptions, so looking to see if a method has declared exceptions isn't going to get you very far. :)

It's up to you on if you catch an exception, but if you're going to catch and return a default value _the least_ you should do is `log.error` or `log.warn` the case. If this method failed for some reason (like NPE, or whatever), it would be phenomenally difficult to detect.

Reply:
> Definitely not the correct approach: Just because a bad pattern is done somewhere else doesn't mean we should spread it elsewhere! We don't do checked exceptions, so looking to see if a method has declared exceptions isn't going to get you very far. :)

[Agreed](https://github.com/spinnaker/orca/pull/3180#discussion_r326297016). I feel like I'm watching myself reviewing my own code. 😄